zwad/download_feature_set.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import argparse
+import hashlib
+import json
+import multiprocessing
+import os
+from itertools import repeat
+from urllib.parse import urljoin
+
+import requests
+
+
+DEST_PATH = '.'
+BASE_API_URL = 'https://zenodo.org/api/records/'
+ZENODO_ID = 4318700
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-j', '--nproc', type=int, help='number of parallel processes')
+    parser.add_argument('-o', '--output', default=DEST_PATH, help='destination path')
+    parser.add_argument('-i', '--id', type=int, default=ZENODO_ID, help='Zenodo ID')
+    args = parser.parse_args()
+    return args
+
+
+def download(description, destdir, session=requests):
+    # zenodo_get maybe a better solution in general:
+    #   https://gitlab.com/dvolgyes/zenodo_get/
+    # but we'd like to practice a bit...
+
+    url = description['links']['self']
+
+    dest = os.path.join(destdir, description['key'])
+    expected_size = description['size']
+
+    # Download resuming is not possible at the moment.
+    # Zenodo doesn't support it. So let's just test the size for equality.
+    # See: https://github.com/zenodo/zenodo/issues/1599
```

Comment:
These comments are copy-paste from some tutorial, or do you want to say something?

Reply:
Ask @balodja